src/core/qgsvectorlayerfeatureiterator.cpp

```diff
@@ -211,12 +211,8 @@ QgsVectorLayerFeatureIterator::QgsVectorLayerFeatureIterator( QgsVectorLayerFeat
   if ( mSource->mHasEditBuffer )
   {
     mChangedFeaturesRequest = mProviderRequest;
-    QgsFeatureIds changedIds;
-    QgsChangedAttributesMap::const_iterator attIt = mSource->mChangedAttributeValues.constBegin();
-    for ( ; attIt != mSource->mChangedAttributeValues.constEnd(); ++attIt )
-    {
-      changedIds << attIt.key();
-    }
+    const QgsFeatureIds changedIds = mSource->mChangedAttributeValues.keys().toSet();
```

Comment:
This is inefficient - you're allocating two lists here. I suspect I probably made this change to fix a clazy warning: https://github.com/KDE/clazy/blob/master/src/checks/level0/README-container-anti-pattern.md